src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -326,9 +329,22 @@ protected virtual void Dispose(bool disposing)
                 }
 
                 _stopEvent.Dispose();
+                _restartHostEvent.Dispose();
 
                 _disposed = true;
             }
         }
+
+        public virtual void Restart()
+        {
+            _restartHostEvent.Set();
+        }
+
+        public virtual void Shutdown()
+        {
+            Stop();
+
+            Process.GetCurrentProcess().Close();
```

Comment:
So in non WebHost scenarios (e.g. a continuous WebJob) this will kill the process, and Kudu will start it up again.